github.js

```diff
@@ -98,20 +57,10 @@ var GithubLocation = function(options, ui) {
   this.versionString = options.versionString + '.1';
 
   // Give the environment precedence over options object
-  if(process.env.JSPM_GITHUB_AUTH_TOKEN) {
-    options.auth = process.env.JSPM_GITHUB_AUTH_TOKEN;
-  } else if (options.username && !options.auth) {
-    options.auth = encodeCredentials(options);
-    // NB deprecate old auth eventually
-    // delete options.username;
-    // delete options.password;
-  }
-
-  if (typeof options.auth == 'string') {
-    this.auth = decodeCredentials(options.auth);
-  }
-  else {
-    this.auth = readNetrc(options.hostname);
+  if (token = process.env.JSPM_GITHUB_AUTH_TOKEN) {
```

Comment:
We now have two tokens to handle - the encoded credentials, and the actual token. A `detectToken` function may come in useful to allow all token fields to be either.

I think it would look like:

```javascript
function isGithubToken(token) {
  return token.match(/[0-9a-f]{40}/);
}
```

Then the old code stays on the `false` path with the new code on the `true` path.